project/{{ cookiecutter.__folder_name }}/.github/workflows/backend.yml

```
@@ -1,124 +1,149 @@
 name: Backend CI
 
 on:
-  push:
-    paths:
-      - "backend/**"
-      - ".github/workflows/backend.yml"
-  workflow_dispatch:
-
-env:
-  IMAGE_NAME_PREFIX: {{ cookiecutter.__container_image_prefix }}
-  IMAGE_NAME_SUFFIX: backend
-  PYTHON_VERSION: "3.11"
+  workflow_call:
+    inputs:
+      base-tag:
+        required: true
+        type: string
+      image-name-prefix:
+        required: true
+        type: string
+      image-name-suffix:
+        required: true
+        type: string
+      python-version:
+        required: true
+        type: string
+      plone-version:
+        required: true
+        type: string
+
+defaults:
+  run:
+    working-directory: backend
 
 jobs:
 
-  meta:
+  lint:
     runs-on: ubuntu-latest
-    outputs:
-      PLONE_VERSION: {{ "${{ steps.vars.outputs.PLONE_VERSION }}" }}
     steps:
-      - name: Checkout
-        uses: actions/checkout@{{ cookiecutter.__gha_version_checkout }}
-
-      - name: Set Env Vars
-        id: vars
-        run: |
-          echo "PLONE_VERSION=$(cat backend/version.txt)" >> $GITHUB_OUTPUT
 
-  black:
+        - name: Checkout
+          uses: actions/checkout@{{ cookiecutter.__gha_version_checkout }}
+
+        - name: Install the latest version of uv
+          uses: astral-sh/setup-uv@{{ cookiecutter.__gha_version_setup_uv }}
+          with:
+            python-version: {{ "${{ inputs.python-version }}" }}
+
+        - name: Check formatting
+          if: {{ "${{ success() || failure() }}" }}
+          id: ruff-format
+          run: uvx ruff@latest format --diff
+
+        - name: Check lint
+          if: {{ "${{ success() || failure() }}" }}
+          id: ruff-lint
+          run: uvx ruff@latest check --diff
+
+        - name: Check XML / ZCML
+          if: {{ "${{ success() || failure() }}" }}
+          id: zpretty
+          run: uvx zpretty@latest --check src
```

@ericof, @ale-rt suggested using the `--include` arg.  If I understand [its readme usage](https://github.com/collective/zpretty?tab=readme-ov-file#usage) correctly, then `--include` limits the files scanned in `src` to only those that end with `.xml` instead of everything in the directory.

For @ale-rt, `uvx` supports [specifying a version of a package](https://docs.astral.sh/uv/guides/tools/#requesting-specific-versions) without needing to call it again. Nice!

```suggestion
          run: uvx zpretty@latest --check --include="\.xml$" src
```

---

AFAICS the default include is already fine.